### PR TITLE
Issue #1491 - add support for automatic converters

### LINKED ISSUE #1491: mpConfig-1.2: support Automatic Converters
Implementations must also support the Automatic converters (eclipse/microprofile-config#269) where there is no corresponding type of converters provided for a given class. The implementation must use the class’s constructor with a single string parameter, then try valueOf(String) followed by parse(CharSequence).

https://github.com/eclipse/microprofile-config/blob/1.2/spec/src/main/asciidoc/converters.asciidoc#automatic-converters

### PR DESCRIPTION
**Pull request by tevans78**
#build 
fixes #1491

### CODE REVIEW

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

File: dev/com.ibm.ws.microprofile.config/src/com/ibm/ws/microprofile/config/converters/DefaultConverters.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+public class DefaultConverters {
+
+    private static final TraceComponent tc = Tr.register(DefaultConverters.class);
+
+    private static final PriorityConverterMap defaultConverters = new PriorityConverterMap();
+
+    static {
+        defaultConverters.addConverter(new IdentityConverter()); // v -> v
+
+        defaultConverters.addConverter(new BooleanConverter());
```
Comment (Emily-Jiang) [inline dev/com.ibm.ws.microprofile.config/src/com/ibm/ws/microprofile/config/converters/DefaultConverters.java]: With the current Config 1.2 implicit converter support, basically there is no need to add the converter implicitly. If no converter is found, it should automatically fallback to ctor, valueOf, parse. In this way, the code is much cleaner.

**Comment by tevans78** [comment]
@neilgsyoung My PB is good enough with just an unrelated CDI failure - https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_VeAswPq8EeefH4uRV76YPQ

**Pull request closed without merging**